### PR TITLE
WIP: Refine the code

### PR DESCRIPTION
**Pull request by Naruto-Sasuke**

### CODE REVIEW

**Comment by Naruto-Sasuke** [comment]
- [x] Simplify the code of `networks.py` a lot.
- [x] Add `init_gain` option.

File: README.md [hunk 1]
```diff
@@ -37,6 +37,10 @@ python test.py
 ```
 The test results will be saved to a html file here: `./results/`.
 
+**Multi-GPU**
+The implementation of `InnerCos` makes it error when using multiple gpus. If you want to try multi-gpu training, please
+set `skip=1` in `options/base_options.py`
```
Comment (Zhaoyi-Yan) [inline README.md]: More notes

File: models/InnerCos.py [hunk 2]
```diff
@@ -26,26 +28,36 @@ def get_target(self):
...
+            # I have to put it here!
+            # when input is image with mask(the second pass), we
+            # Mention only when input is the groundtruth, the target makes sense.
+            self.target = in_data.narrow(1, self.c // 2, self.c // 2).clone() # the latter part
+            self.target = self.target * self.strength
+            self.target = self.target.detach()
+
+            self.output = in_data
+        else:
+            self.output = in_data
```
Comment (Zhaoyi-Yan) [inline models/InnerCos.py]: Here, you should also define `self.loss`.

**Comment by Naruto-Sasuke** [comment]
Done.
